### PR TITLE
chore(git): Make builder git remote use different hostname

### PR DESCRIPTION
**Pull request by krancour**
This transitions the CLI from creating git `deis` git remotes that use the same hostname as the controller to using one with a hostname _derived_ from the controller's hostname.

This is to support the contingency wherein some scenario has forced an admin to use _two_ different load balancers-- one for builder and another for everything else.

See https://github.com/deis/controller/issues/657 for further context.

### CODE REVIEW

File: pkg/git/git_test.go [hunk 1]
```diff
@@ -8,14 +8,14 @@ func TestRemoteURL(t *testing.T) {
 	t.Parallel()
 
 	actual := RemoteURL("example.com", "app")
-	expected := "ssh://git@example.com:2222/app.git"
+	expected := "ssh://git@example-builder.com:2222/app.git"
```
Comment (bacongobbler) [inline pkg/git/git_test.go]: this doesn't seem right to me. So if someone wanted to expose the builder on example.com, they'd have to own both example.com and example-builder.com?

Comment (krancour) [inline pkg/git/git_test.go]: @bacongobbler, I think I mindlessly updates that test case without realizing it was at least an _edge_ case, or possibly even an _invalid_ case.

In the typical case, the controller would be at the `deis.example.com`, which, according to this logic, would make `deis-builder.example.com` the subdomain for the builder.

I'm trying to think of a scenario where the controller _isn't_ exposed as some subdomain of the platform domain.  From the router's perspective, the controller is "just another app," which is named `deis`, so it automatically occupies the `deis` subdomain of the platform domain (e.g. `example.com`).  Like other apps, it _could_ have a custom / vanity URL such as `mydeis.com`, but that's only achieved through some fairly major modifications to the charts...

I'm debating whether this edge case is worth addressing vs whether is is _so "edgy"_ that I should just update this test case to be more realistic.

Thoughts?

Comment (bacongobbler) [inline pkg/git/git_test.go]: Yeah I don't think this edge case is worth addressing since we don't route on TLDs. I think it's best to just update the unit tests to `deis.example.com` instead of `example.com` so others don't get tripped up by this either.

**Comment by krancour** [comment]
@bacongobbler updated as we discussed.
